docs/preview/logdecorations.md

```diff
@@ -0,0 +1,260 @@
+
+# Log decorations
+
+
+Task authors should be able to control how the log output is displayed to the end user.
+This outlines different decoration options that are available.
+
+## Special lines
+* Errors
+  * `##[error] I am an error` 
+* Warnings
+  * `##[warning] I am a warning` 
+* Commands
+  * `##[warning] I am a command/a tool` 
+* Sections
+  * `##[section] I am a section, which is usually whole task step. I am not typically used.` 
+
+## Collapse
+
+>This is a draft spec and is subject to change
+
+>Note that that if you log an error using ```##vso[task.logissue]error/warning message``` command (see [logging commands](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/docs/authoring/commands.md) here) we will surface those errors in build view and when clicked , we will automatically jump to that particular line. If it's already part of a group, we will auto-expand the group.
+
+Task authors can mark any part of the log as a collapsible region using these decorations:
+
+Starting the collapsible region - `##[startgroup<optionalGroupName>]`
```

Comment:
`##[startgroup] name` for consistency with existing decorator format

Reply:
I kinda feel like if it's `##[startgroup] name` it implies that grouping happens with title as `name`
However, it's just a marker not group title.
Group title is just the first line after the group decoration, I think `##[startgroup:name]` will make that clear a bit.

Reply:
i see, i thought it was the title. If not, why add the concept at all?